### PR TITLE
add history to several models

### PR DESCRIPTION
**Pull request by jansenk**
add history to

- Curriculum
- CurriculumProgramMembership
- CurriculumCourseMembership
- DegreeDeadline
- DegreeCost

### CODE REVIEW

File: course_discovery/apps/course_metadata/migrations/0159_historicalcurriculum_historicalcurriculumcoursemembership_historicalcurriculumprogrammembership_hist.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
```
Comment (iloveagent57) [inline course_discovery/apps/course_metadata/migrations/0159_historicalcurriculum_historicalcurriculumcoursemembership_historicalcurriculumprogrammembership_hist.py]: Could you make this filename a wee bit shorter? :)

File: course_discovery/apps/publisher/migrations/0075_auto_20190213_2015.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
```
Comment (iloveagent57) [inline course_discovery/apps/publisher/migrations/0075_auto_20190213_2015.py:1]: I'm guessing that this was generated when upgrading the `simple-history` version, correct?  Any idea what exactly the purpose of this migration is?

**Review by iloveagent57 — COMMENTED** [review]
The changes in `course_metadata` generally look good, I just have a question about the publisher migration.

File: course_discovery/apps/course_metadata/migrations/0159_historicalcurriculum_historicalcurriculumcoursemembership_historicalcurriculumprogrammembership_hist.py (hunk 1 above)
Comment (jansenk) [reply course_discovery/apps/course_metadata/migrations/0159_historicalcurriculum_historicalcurriculumcoursemembership_historicalcurriculumprogrammembership_hist.py]: Can I just rename it?

File: course_discovery/apps/publisher/migrations/0075_auto_20190213_2015.py (hunk 2 above)
Comment (jansenk) [reply course_discovery/apps/publisher/migrations/0075_auto_20190213_2015.py:1]: You're correct, it seems to be due to the upgrade. I compared it tothe current definition of these columns and it looks like it is adding default=None to the historical fields to match the fields on the base models

**Comment by jansenk** [comment]
@iloveagent57 any other comments?

**Review by schenedx — APPROVED** [review]
👍

**Pull request merged**